src/Microsoft.DotNet.Darc/src/DarcLib/ILocalGitRepo.cs

```diff
@@ -45,4 +46,21 @@ public interface ILocalGitRepo : IGitRepo
     /// <param name="repoDir">Path to a git repository</param>
     /// <returns>List of currently modified staged files</returns>
     IEnumerable<string> GetStagedFiles(string repoDir);
+
+    /// <summary>
+    /// Pushes a branch to a remote
+    /// </summary>
+    /// <param name="repoPath">Path of the local repository</param>
+    /// <param name="remoteName">Name of remote to push to</param>
+    /// <param name="branchName">Name of branch to push</param>
+    /// <param name="identity">Identity object containing username and email</param>
+    /// <param name="gitHubPat">Token to authenticate for pushing to GitHub</param>
+    /// <param name="azureDevOpsPat">Token to authenticate for pushing to AzureDevOps</param>
+    void Push(
+        string repoPath, 
+        string remoteName, 
+        string branchName, 
+        LibGit2Sharp.Identity identity, 
+        string gitHubPat = null, 
+        string azureDevOpsPat = null);
```

Comment:
Maybe we should just pass in a token, whatever the remote is?

We could decide on which one level above and leave this agnostic and more generic?

Reply:
that is better for sure but to get the repo type from the remote we have to create a Repository object again